### PR TITLE
fix(ingest/redshift):fixing schema filter

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/redshift/config.py [hunk 1]
```diff
@@ -137,3 +146,22 @@ def check_database_or_database_alias_set(cls, values):
...
+    def backward_compatibility_configs_set(cls, values: Dict) -> Dict:
+        match_fully_qualified_names = values.get("match_fully_qualified_names")
+
+        schema_pattern: Optional[AllowDenyPattern] = values.get("schema_pattern")
+
+        if (
+            schema_pattern is not None
+            and schema_pattern != AllowDenyPattern.allow_all()
+            and match_fully_qualified_names is not None
+            and not match_fully_qualified_names
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/redshift/config.py:160]: `match_fully_qualified_names is False` seems simpler

File: metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py [hunk 2]
```diff
@@ -772,10 +780,12 @@ def gen_dataset_workunits(
...
-            if self.config.schema_pattern.allowed(f"{database}.{schema}"):
-                logging.debug(
-                    f"Not caching tables for schema {database}.{schema} which is not allowed by schema_pattern"
-                )
+            if is_schema_allowed(
+                self.config.schema_pattern,
+                schema,
+                database,
+                self.config.match_fully_qualified_names,
+            ):
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/redshift/redshift.py:788]: Can we keep this logic consistent with the above -- handle the `not` case first and `continue`? Same for above

**Review by asikowitz — APPROVED** [review]
Just some style comments, LGTM. Was the bug entirely just how schema patterns are matched?

**Review by hsheth2 — APPROVED** [review]
LGTM

**Pull request merged**